public/app/plugins/datasource/grafana-azure-monitor-datasource/credentials.ts

```diff
@@ -116,7 +115,17 @@ export function getCredentials(options: AzureDataSourceSettings): AzureCredentia
   }
 }
 
-export function getLogAnalyticsCredentials(options: AzureDataSourceSettings): AzureCredentials | undefined {
+export function credentialsDiffer(logCreds: AzureClientSecretCredentials, primaryCreds: AzureClientSecretCredentials) {
```

Comment:
I would keep isLogAnalyticsSameAs and use it to check whether createntials are different or not.
In frontend it's impossible to check equality of secret, and "azureLogAnalyticsSameAs" should be consistent.